src/hex_tarball.erl

```diff
@@ -0,0 +1,455 @@
+-module(hex_tarball).
+-export([create/2, unpack/2, format_checksum/1, format_error/1, gzip/1, normalize_requirements/1]).
+-define(VERSION, <<"3">>).
+-define(TARBALL_MAX_SIZE, 8 * 1024 * 1024).
```

Comment:
in a separate PR we'll make tarball max size configurable via options to create and unpack